src/pages/references/node-configuration.md

```diff
@@ -0,0 +1,489 @@
+
+Start a node with a config of your own. Can be used for joining a network, starting new chain, etc.
+
+#### Arguments
+
+**--config**: relative or absolute path to the TOML config file. Required.
+
+Example:
+
+```bash
+stacks-node start --config=/path/to/config.toml
+```
+
+See [Configuration File Options](#configuration-file-options) for more information.
+
+#### version
+
+Displays information about the current version and the release cycle.
+
+Example:
+
+```bash
+stacks-node version
+```
+
+#### help
+
+Displays a help message.
+
+Example:
+
+```bash
+stacks-node help
+```
+
+## Configuration File Options
+
+The TOML configuration file has multiple sections under which an option may be placed.
+
+To see a list of example configurations, [please see this page](https://github.com/blockstack/stacks-blockchain/tree/master/testnet/stacks-node/conf).
+
+### Section: node
+
+Contains various configuration options pertaining to the stacks-node.
+
+Example:
+
+```toml
+[node]
+rpc_bind = "0.0.0.0:20443"
+p2p_bind = "0.0.0.0:20444"
+bootstrap_node = "048dd4f26101715853533dee005f0915375854fd5be73405f679c1917a5d4d16aaaf3c4c0d7a9c132a36b8c5fe1287f07dad8c910174d789eb24bdfb5ae26f5f27@testnet-miner.blockstack.org:20444"
+# Enter your private key here!
+seed = "replace-with-your-private-key"
+miner = true
+```
+
+#### working_dir (optional)
+
+Absolute path to the directory which the stacks-node will use for storing various data.
+
+Example:
+
+```toml
+working_dir = "/root/stacks-node"
+```
+
+#### rpc_bind
+
+Address and port stacks-node should bind to for RPC connections.
+
+Example:
+
+```toml
+rpc_bind = "0.0.0.0:20443"
+```
+
+#### p2p_bind
+
+Address and port stacks-node should bind to for P2P connections.
+
+Example:
+
+```toml
+p2p_bind = "0.0.0.0:20444"
+```
+
+#### data_url (optional)
+
+Address and port from which the stacks-node will be receiving incoming rpc connections.
+
+Example:
+
+```toml
+data_url = "1.2.3.4:20443"
+```
+
+#### p2p_address (optional)
+
+Address and port from which the stacks-node will be receiving incoming p2p connections.
+
+Example:
+
+```toml
+p2p_address = "1.2.3.4:20444"
+```
+
+#### bootstrap_node (optional)
+
+Public key, address, and port stacks-node should use to pull transaction data from when starting.
+
+Example:
+
+```toml
+bootstrap_node = "048dd4f26101715853533dee005f0915375854fd5be73405f679c1917a5d4d16aaaf3c4c0d7a9c132a36b8c5fe1287f07dad8c910174d789eb24bdfb5ae26f5f27@krypton.blockstack.org:20444"
+```
+
+#### wait_time_for_microblocks (optional)
+
+Time to wait (in milliseconds) for microblocks.
+
+Example:
+
+```toml
+wait_time_for_microblocks = 15000
+```
+
+#### seed (optional)
+
+The private key to use for mining. Only needed if `miner` is set to `true`.
+
+[See this page for information on how to generate a private key.](https://docs.blockstack.org/mining)
+
+Example:
+
+```toml
+seed = "replace-with-your-private-key"
+```
+
+#### local_peer_seed (optional)
+
+The private key to use for mining.
```

Comment:
What is the difference to `seed`?

Reply:
Not sure, @kantai or @lgalabru ?

Reply:
This is the seed used for signing P2P messages in the networking stack, it differentiates network peers (and is used even by non-mining nodes).